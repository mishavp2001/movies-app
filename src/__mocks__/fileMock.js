/**
 * @providesModule EmptyModule1
 */
module.exports = '';
