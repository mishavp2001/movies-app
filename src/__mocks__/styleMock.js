/**
 * @providesModule EmptyModule2
 */
module.exports = '';
