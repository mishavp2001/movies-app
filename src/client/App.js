import React, { Component } from "react";
import Gallery from "./components/Gallery";
import ReactImage from "./react.png";
import socketIOClient from "socket.io-client";
import "./css/app.css";

const DEFAULT_SORT = "movieId";

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = { movies: null, sortedBy: DEFAULT_SORT };
  }

  componentDidMount() {
    //API call
    this.getMovies();
  }

  getMovies() {
    const socket = socketIOClient('http://localhost:8080/');
    socket.on('connect', () => {
      //console.log("Socket Connected");
      socket.emit('movies', "Load");
      socket.on("movies", movies => {
        //Time out to show asynchronious behavier
        setTimeout(() => { this.setState({ movies: movies })}, 4000);
        //this.setState({ movies: movies });
      });
    });
    socket.on('disconnect', () => {
      socket.off("movies")
      socket.removeAllListeners("movies");
      //console.log("Socket Disconnected");
    });
  }

  sortBy = (sort, e) => {
    //Call back to set state to sorted movies by sort parameter
    e.preventDefault();
    e.stopPropagation();
    if(!sort){return false};
    const moviesSorted = this.state.movies.sort((a, b) => (a[sort] > b[sort]) ? 1 : ((b[sort] > a[sort]) ? -1 : 0))
    this.setState(
      {movies: moviesSorted, sortedBy: sort}
    )
  }

  render() {
    return (
      <div className="gallery-container">
        <h1>Welcome to our gallery!</h1>
        {this.state.movies ? (
          <Gallery moviesResp={this.state.movies} sortedBy={this.state.sortedBy} sortBy={this.sortBy}/>
        ) : (
          <h1>Loading.. please wait!</h1>
        )}
      </div>
    );
  }
}
