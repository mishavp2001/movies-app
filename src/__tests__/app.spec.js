jest.mock('../__mocks__/socketIOClient');

import React from 'react'
import { shallow, mount } from 'enzyme';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer'
import {serverSocket} from '../__mocks__/socketIOClient';
import App from '../client/App.js'

enzyme.configure({ adapter: new Adapter() });

function tick() {
  return new Promise(resolve => {
    setTimeout(resolve, 0);
  })
}


// Snapshot for CardComponent React Component
describe('>>>App --- Snapshot',()=>{
    it('+++capturing Snapshot of App', () => {
        const renderedValue =  renderer.create(<App />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});

describe('App renders', ()=>{
    let wrapper;
    beforeEach(() => {
      wrapper = mount(<App />);
    });

    it('renders app with no props without waiting for movies response', async () => {
        await expect(wrapper.find(".gallery-container")).toHaveLength(1);
        await expect(wrapper.find("h1").first().text()).toEqual("Welcome to our movies gallery!");
        await expect(wrapper.find("h1").at(1).text()).toEqual("Loading.. please wait!");
    });

    it('renders initial state', async () => {
        wrapper.setState({movies: [{id: 11212}]});
        await expect(wrapper.find("Gallery")).toHaveLength(1);
    });

    it('renders one movie', async () => {
      wrapper.setState({movies: [{id: 11212}]});
      await expect(wrapper.find("GalleryImage")).toHaveLength(1);
      expect(wrapper.find("img").prop('alt')).toEqual("Image number 1");
    });

    it('Calls getMovies on render ', async () => {
      const spy = jest.spyOn(App.prototype, 'getMovies');
      const temp = mount(<App />);
      await expect(spy).toHaveBeenCalled();
    });
});

describe('App interaction', ()=>{
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<App />);
  });

  it('Sorts by sortBy method', async () => {
    const spy = jest.spyOn(wrapper.instance(), 'sortBy');
    wrapper.setState({movies: [{id: 1}, {id: 3}, {id: 2}]});
    await expect(wrapper.find("GalleryImage")).toHaveLength(3);
    expect(wrapper.find(".sortLink")).toHaveLength(2);
    //Sorts by firs link
    wrapper.find(".sortLink").first().simulate('click');
    expect(spy).toHaveBeenCalled();
    //Sorts by second link
    wrapper.find(".sortLink").at(1).simulate('click');
    expect(spy).toHaveBeenCalled();
  });
  it('Clicking on card opens/closes modal overlay', async () => {
    const spy = jest.spyOn(wrapper.instance(), 'sortBy');
    wrapper.setState({movies: [{id: 1}, {id: 3}, {id: 2}]});
    await expect(wrapper.find("GalleryImage")).toHaveLength(3);
    expect(wrapper.find(".sortLink")).toHaveLength(2);
    //Open modal
    wrapper.find(".card-icon-open").first().simulate('click');
    expect(wrapper.find(".modal-overlay")).toHaveLength(1);
    //Close Modal
    wrapper.find(".modal-close").first().simulate('click');
    expect(wrapper.find(".modal-overlay")).toHaveLength(0);
  });
});
