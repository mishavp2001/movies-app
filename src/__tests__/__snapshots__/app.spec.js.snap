// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>App --- Snapshot +++capturing Snapshot of App 1`] = `
<div
  className="gallery-container"
>
  <h1>
    Welcome to our movies gallery!
  </h1>
  <h1>
    Loading.. please wait!
  </h1>
</div>
`;
